src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ConstructedType.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.Runtime;
+using Internal.TypeSystem;
+using Internal.IL;
+using Internal.Text;
+
+using ILCompiler.DependencyAnalysisFramework;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class ConstructedType : DependencyNodeCore<NodeFactory>, IEETypeNode
```

Comment:
Having a distinct constructed and unconstructed types is useful for size optimizations in fully AOT compiled applications with a whole program view at compile time. How is this going to work if we have a JIT and no whole program view?